private/buf/buflsp/file.go

```diff
@@ -307,12 +279,19 @@ func (f *file) IndexSymbols(ctx context.Context) {
 
 	// Throw away all the old symbols and rebuild symbols unconditionally. This is because if
 	// this file depends on a file that has since been modified, we may need to update references.
-	f.symbols = nil
-	f.referenceSymbols = nil
-	f.referenceableSymbols = make(map[string]*symbol)
+	clear(f.symbols)
+	f.symbols = f.symbols[:0]
+	clear(f.referenceSymbols)
+	f.referenceSymbols = f.referenceSymbols[:0]
+	clear(f.referenceableSymbols)
+	if f.referenceableSymbols == nil {
+		f.referenceableSymbols = make(map[string]*symbol)
+	}
```

Comment:
is this actually fixing something, or are we just trying to assert here that we can definitely gc the old values? feels like the old version is clearer and less reliant on knowing the tricks to make gc run?